### ISSUE TITLE
Reactions remain unread

### ISSUE THREAD

**Issue by Peter-maximus** [tomer8007/whatsapp-web-incognito#140]
If "Don't send read confirmations" and "Auto-send receipts on reply" is both turned on, it works correctly for messages, however not for reactions (those smileys etc.).
I mean that if i reply in the chat after reaction, **the reaction reamains unread..**.

It seems that web whatsapp with incognito plugin doesn't show reaction state (read / unread), however whatsapp app in mobile takes it as unread (even if i reply on it in web whatsapp with incognito plugin) and once new message come it shows popup of new message together with previous unread reactions.
At the end if you receive number of reactions from different chats and read them all just on web whatsapp with incognito plugin and afterwards come just one new regular message then whatsapp app in mobile shows you all that reactions again as new repeatedly.

Only workaround now is to make those reaction read in whatsapp app in mobile.

**Comment by tomer8007** [comment]
I will try to fix this

**Comment by Peter-maximus** [comment]
thx a lot

**Comment by tomer8007** [comment]
This bug is due to cb01c48e0133e3dd18295537968b65a4732768d7 which I commited last year because it addressed another issue which I can't remember (it had something to do with read receipt privacy settings). 

Anyway I now reverted it in c9fc8f4e0b8585ad0c099de0c17d765cdcb23037 and if nothing bad will happen this will get into the next version in the store.

**Comment by Peter-maximus** [comment]
thx

**Comment by tomer8007** [comment]
@Peter-maximus I think that for groups that are part of communities, a `read-self` is sent, so clicking on such a chat in this mode marks it as read.
Are you aware of a feature in communities where you can't see who read your message?

**Comment by Peter-maximus** [comment]
apologize, i do not use group chats.

however do i understand correctly that you are saying that group chats are automatically marked as read immediatelly once i click on a chat (even if "don't send read confirmations" are enabled in plugin?)

**Comment by tomer8007** [comment]
@Peter-maximus Yes, but I'm saying it happens only in groups that are part of WhatsApp communities and only if the "fix" is on - blocking `read-self` receipts.

**Comment by Peter-maximus** [comment]
understand.

so for me no issue as i don't use them at all.

**Comment by Peter-maximus** [comment]
any update on this?
**the reactions remain unread...**

**Comment by tomer8007** [comment]
Yes, This is because I saw that when I make the reactions read, it makes a side effect where it looks like messages in communities are marked as read when you click on them, even when the blocking is on.

If you don't care about this, you can fix it by removing one line of code:
https://github.com/tomer8007/whatsapp-web-incognito/blob/ae3b46b0f66e2b6309a73d311c1fae9eeb55d98b/core/interception.js#L231

**Comment by Peter-maximus** [comment]
would you mind to add it into GUI?

just additional simple checkbox in waincognito options popup menu.

 

especially as i don't know how to compile that amended code and secondly it will be also not automatically updated if new version is released…

**Comment by tomer8007** [comment]
How would you call such an option?

**Comment by Peter-maximus** [comment]
i think about following:

 

Name of the option:

.         Send read confirmations for reactions (on / off)

 

Description below the option (like all other options already have):

.         By enabling this option also messages in communities are marked as read when you click on them.

**Comment by tomer8007** [comment]
OK, well, I don't know if it's important enough, maybe in the future... What do you think about a javascript variable in the meantime?

**Comment by Peter-maximus** [comment]
I would say it's very important as without it you have the "reaction" messages sticked in the notification bar on phone.

Additionally the phone behaves as there is unread message waiting for you (diode blinking, regular sound reminder etc.), even you have read it on computer.

So the whole goal of using computer instead of phone for chat loosing sense this way.

 

What do you mean by java script variable way?

**Comment by tomer8007** [comment]
I mean that, you could open open the console (F12), and type, say, `shouldSendReactionsReceipts = true`, and from this moment until you close the window on the extension will not block receipts for reactions.

**Comment by Peter-maximus** [comment]
that is very unconvenient, as it means every time i open whatsapp window i have to open console and write this command...

**i think it is important to have this option within program GUI** otherwise reaction messages left sticked in the notification bar on the phone with all the consequences mentioned above.
would you mind to implement it please?
thx a lot.

**Comment by tomer8007** [comment]
OK, I might implement that after I'll fix some more important [bugs](https://github.com/tomer8007/whatsapp-web-incognito/issues/146).

**Comment by Peter-maximus** [comment]
thx a lot again. have a nice evening
